core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -105,6 +116,41 @@ protected static void onRetryingComplete(ClientRequestContext ctx) {
         return retryStrategy;
     }
 
+    /**
+     * Removes a retry {@link ScheduledFuture} which is added in
+     * {@link #scheduleNextRetry(ClientRequestContext, Consumer, Runnable, long)}.
+     */
+    protected void removeRetryScheduleIfExist(ClientRequestContext ctx) {
+        final Entry<ScheduledFuture<Void>, FutureListener<Void>> futureAndListener = retrySchedules.remove(ctx);
+        if (futureAndListener != null) {
+            assert ctx.eventLoop().inEventLoop() : "Cannot remove listener in another thread: " +
+                                                   Thread.currentThread().getName();
+
+            final ScheduledFuture<Void> future = futureAndListener.getKey();
+            final FutureListener<Void> lister = futureAndListener.getValue();
+            future.removeListener(lister);
```

Comment:
You don't have to remove the listener, because the listener list is cleared after notification: https://github.com/netty/netty/blob/4.1/common/src/main/java/io/netty/util/concurrent/DefaultPromise.java#L470

Reply:
Oops! I didn't know. Thank you!